src/language/patterns.md

```diff
@@ -0,0 +1,334 @@
+---
+title: Patterns
+description: Summary of patterns in Dart.
+---
+
+{{site.alert.version-note}}
+  Patterns require a [language version][] of at least 3.0.
+{{site.alert.end}}
+
+Patterns are a syntactic category in the Dart language, like statements and expressions.
+They represent the form of a value, not an actual value.
+
+This page describes:
+- What patterns do (match and destructure)
+- Where patterns are allowed in Dart code
+- Common use cases for patterns
+
+To learn about the different types of patterns, visit the [pattern type reference][types].
+
+## Matching
+
+A pattern is always tested against a value to determine if the value has the form
+you expect, or in other words, if the value _matches_ the pattern. 
+
+What constitutes a match depends on the [pattern type][types].
+For example, matching a constant pattern means the value is equivalent to the pattern:
```

Comment:
I don't know what "the value is equivalent to the pattern" means here.  What is this sentence trying to say?  Something like "For example, the following switch usess a constant pattern to test that `number` is `1`?  I'm not sure.